core/src/main/java/org/infinispan/marshall/core/UserExternalizerObjectTable.java

```diff
@@ -0,0 +1,57 @@
+package org.infinispan.marshall.core;
+
+import static org.infinispan.marshall.core.GlobalMarshaller.ID_EXTERNAL;
+import static org.infinispan.marshall.core.GlobalMarshaller.writeExternalClean;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.function.BiConsumer;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.jboss.marshalling.ObjectTable;
+import org.jboss.marshalling.Unmarshaller;
+
+/**
+ * A {@link org.jboss.marshalling.ObjectTable} implementation that creates {@link org.jboss.marshalling.ObjectTable.Writer}
+ * based upon a users configured {@link org.infinispan.marshall.core.MarshalledEntryImpl.Externalizer} implementation.
+ */
+class UserExternalizerObjectTable implements ObjectTable {
+   final ClassToExternalizerMap exts;
+   final ClassToExternalizerMap.IdToExternalizerMap reverseExts;
+
+   UserExternalizerObjectTable(ClassToExternalizerMap exts) {
+      this.exts = exts;
+      this.reverseExts = exts.reverseMap();
+   }
+
+   @Override
+   public ObjectTable.Writer getObjectWriter(Object object) {
+      BiConsumer<ObjectOutput, Object> writer = findWriter(object);
+      return writer != null ? writer::accept : null;
+   }
+
+   @Override
+   public Object readObject(Unmarshaller unmarshaller) throws IOException, ClassNotFoundException {
+      AdvancedExternalizer<Object> ext = findExternalizerIn(unmarshaller);
+      return ext.readObject(unmarshaller);
+   }
+
+   BiConsumer<ObjectOutput, Object> findWriter(Object object) {
```

Comment:
I say lets just return the `AdvancedExternalizer` instead. Then we only need 1 lambda since `getObjectWriter` can change to `return ext != null ? (m, obj) -> writeExternalClean(obj, ext, m);` or something like that.